src/binder_voicecall.c

```diff
@@ -1768,29 +1967,62 @@ binder_voicecall_ecclist_changed(
 {
     BinderVoiceCall* self = user_data;
     GBinderReader reader;
-    gsize count = 0;
-    const RadioEmergencyNumber* list;
     const char** en_list = NULL;
 
     /* currentEmergencyNumberList(RadioIndicationType,vec<EmergencyNumber>); */
     gbinder_reader_copy(&reader, args);
-    list = gbinder_reader_read_hidl_type_vec(&reader,
-        RadioEmergencyNumber, &count);
-
-    DBG_(self, "%u emergency number(s)", (guint) count);
-    if (count) {
-        guint i;
-
-        en_list = g_new(const char*, count + 1);
-        for (i = 0; i < count; i++) {
-            en_list[i] = list[i].number.data.str;
-            DBG("%s", en_list[i]);
+    if (self->interface_aidl == RADIO_AIDL_INTERFACE_NONE) {
+        gsize count = 0;
+        const RadioEmergencyNumber* list;
+        list = gbinder_reader_read_hidl_type_vec(&reader,
+            RadioEmergencyNumber, &count);
+
+        DBG_(self, "%u emergency number(s)", (guint) count);
+        if (count) {
+            guint i;
+
+            en_list = g_new(const char*, count + 1);
```

Comment:
Again, to avoid the last case to terminate this just use `g_new0()`.

Reply:
Done